src/client/Microsoft.Identity.Client/AppConfig/AuthorityInfo.cs

```diff
@@ -408,6 +416,11 @@ internal static string GetSecondPathSegment(Uri authority)
 
         private static AuthorityType GetAuthorityType(string authority)
         {
+            if (isCiamAuthority(authority))
```

Comment:
This leads to double parsing of the string via the Uri constructor, which is expensive (this is on the critical path). I recommend revamping this logic a bit - instead of using `GetFirstPathSegment` do the following: 

1. `(host, firstPathSegment, secondPathSegment) = ParseAuthority(Uri)` - this ensures that we only parse this once. 
2. [Optional, but recommended] Ensure that internally we use exclusively `URI` for the authority. `new Uri(string)` is an expensive operation that needs to be avoided. Not sure how involved this is, seems doable, but the devil is in the details.

Reply:
I did a bit of refactoring to reduce the overall number of new Uri() calls. Please take a look and verify. There were a number of cases where we seem to be creating new Uris excessivly and it didnt make sense to me why some of them were there in the first place.